src/gen_smtp_server_session.erl

```diff
@@ -1405,6 +1416,32 @@ report_recipient(multiple, [{ResponseType, Value} | Rest], State) ->
     report_recipient(ResponseType, Value, State),
     report_recipient(multiple, Rest, State).
 
+% parse proxy protocol
+% http://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
+
+read_proxy_protocol(Transport, Socket) ->
+    ok = Transport:setopts(Socket, [{packet, line}, binary]),
+    case Transport:recv(Socket, 0, 5000) of
+        {ok, Data} ->
+            case Data of
```

Comment:
There's also a binary form of this protocol (version 2), were you planning on supporting this as well?

Reply:
We can support if you think it makes sense. Idea is that you have control in what version you want to use:

`send-proxy` is v1 (current implementation) 
`send-proxy-v2` - is v2